home/.chezmoiscripts/universal/run_before_01-import-secrets-envchain.tmpl

```diff
@@ -0,0 +1,31 @@
+{{- if (ne .host.distro.family "windows") -}}
+#!/usr/bin/env bash
+# @file Import environment variable values into `envchain`
+# @brief Ensures that values of sensitive environment variables are imported into `envchain` on either Linux or macOS
+# @description
+#     This script imports the values of sensitive environment variables into `envchain` if it is installed
```

Comment:
Hey, can you add a bit more detail here on how it works?

Also, maybe we should cycle through the variables in the .chezmoitemplates/secrets folder instead to load them into envchain. That way the secrets are not stored anywhere on the disk in unencrypted format. We can get rid of the `private.sh` file (unless you think it's better this way then let me know what you think).

Reply:
Not having the keys in unencrypted form is better from a security standpoint. I saw in the documentation (or in the script templates) where we ask to source `private.sh`. We would need to update those instances to instead use `envchain <namespace> <executable>`.

A couple pain points:

* Each execution of the command needs to be prefixed with `envchain <namespace>`. Creating aliases may help with that. We can also provide instructions to set the variables from a namespace in the current shell
* The file also gives an easy way to load all values in one go. I do not think there is a way to load all the variables using `envchain`. This makes available what is absolutely needed to execute the program, which is good. Could we disable creation of `private.sh` unless opted for?

To add the secrets to envchain from the files, we need to name the files in such a way that we can detect the namespace from the name of the file. The part till the first underscore character can be taken as the name of the namespace and the entire file name can be the variable name. Existing files would work if we are to switch to the above method; some files need updating, e.g. `key-cloudflare-api-key`, `key-slack-app-client-id`, etc..

Reply:
Hey @enggnr -- let's replace `private.sh` as an alias for `envchain` source import. Then, we can move all the variable declarations into a script file that lives right by the `private.sh` that imports all the variables into envchain. I think this is better than automatically importing all the variables from `.chezmoitemplates/secrets`.

Just import everything into the default collection.

Reply:
Be sure to include documentation in both the `private.sh` file and the helper `after_script_script-name.sh.tmpl` file.